corehq/util/context_processors.py

```diff
@@ -71,15 +71,19 @@ def domain(request):
 
 def domain_billing_context(request):
     is_domain_billing_admin = False
+    restrict_domain_creation = settings.RESTRICT_DOMAIN_CREATION
     if getattr(request, 'couch_user', None) and getattr(request, 'domain', None):
         account = BillingAccount.get_account_by_domain(request.domain)
         if account:
             if request.couch_user.username in account.enterprise_admin_emails:
                 is_domain_billing_admin = True
             elif has_privilege(request, privileges.ACCOUNTING_ADMIN):
                 is_domain_billing_admin = True
+            if not is_domain_billing_admin:
+                restrict_domain_creation = account.restrict_domain_creation
```

Comment:
This seems like it will affect cases even where `settings.RESTRICT_DOMAIN_CREATION == True`, which might not make PNA happy, or might require some more configuration/migration. Maybe `settings.RESTRICT_DOMAIN_CREATION or account.restrict_domain_creation`?

Reply:
good point, will update